src/ESLint.Formatter/sarif.js

```diff
@@ -145,12 +152,29 @@ module.exports = function (results, data) {
                         }
 
                         if (sarifRuleIndices[message.ruleId] !== "undefined") {
-                            sarifResult.ruleIndex = sarifRuleIndices[message.ruleId];
+                            sarifRepresentation.ruleIndex = sarifRuleIndices[message.ruleId];
+                        }
+                    } else {
+                        // ESLint produces a message with no ruleId when it encounters an internal
+                        // error. SARIF represents this as a tool execution notification rather
+                        // than as a result, and a notification has a descriptor.id property rather
+                        // than a ruleId property.
+                        sarifRepresentation.descriptor = {
+                            id: internalErrorId
```

Comment:
>internalErrorId [](start = 32, length = 15)

What kind of internal errors are we talking about? If the internal error is something like: you passed us code to analyze that isn't syntactically correct, that would be a user configuration error, not an internal error. Some tools seem to mix these concepts together. A tool notification error is intended to convey 'we should have been able to scan for you, but an issue in the engine itself prevented us from succeeding'. A configuration error is 'we received inputs or configuration by the user which prevented us from completing a scan'.

Reply:
I'm still researching the root cause, but it appears to be ESLint's parsing engine (specifically, a component called the "scope analyzer") choking on a construct like this:
```
import Example from 'example';
```
This is valid starting in ES 2015. You could argue that this is a configuration error because the ESLint user didn't set that option. But

1. I'm still investigating, and
2. Even if that turned out to be the case for this particular error, it's not clear how the SARIF generator could look at an ESLint message-without-a-ruleId and determine in general whether it's a tool failure or a configuration error.

I mean, maybe I can actually look at the message and, if I recognize the string as a known configuration issue, I can put it in the tool configuration notifications, otherwise in the tool execution notifications.

But it's still early in my research. Don't take any of what I just said too seriously yet.

This issue was reported in November 2015 in [[https://github.com/eslint/eslint/issues/4344](https://github.com/eslint/eslint/issues/4344)](https://github.com/eslint/eslint/issues/4344).


---
In reply to: [385956543](https://github.com/microsoft/sarif-sdk/pull/1793#discussion_r385956543) [](ancestors = 385956543)

Reply:
Per our discussion: we will record these as tool configuration notifications for now, and record the question of how to distinguish tool configuration notifications from tool execution notifications in [https://github.com/Microsoft/sarif-sdk/issues/1798](https://github.com/Microsoft/sarif-sdk/issues/1798). @mikefan@microsoft.com FYI.

---
In reply to: [385975793](https://github.com/microsoft/sarif-sdk/pull/1793#discussion_r385975793) [](ancestors = 385975793,385956543)